components/script/dom/offscreencanvas.rs

```diff
@@ -142,6 +147,85 @@ impl OffscreenCanvas {
     }
 }
 
+impl Transferable for OffscreenCanvas {
+    type Index = OffscreenCanvasIndex;
+    type Data = TransferableOffscreenCanvas;
+
+    fn can_transfer(&self) -> bool {
+        // TODO (#37882): Allow to transfer with a placeholder canvas element.
+        self.placeholder.is_none()
+    }
+
+    /// <https://html.spec.whatwg.org/multipage/#the-offscreencanvas-interface:offscreencanvas-16>
+    fn transfer(&self) -> Result<(OffscreenCanvasId, TransferableOffscreenCanvas), ()> {
+        // Step 1. If value's context mode is not equal to none, then throw an
+        // "InvalidStateError" DOMException.
+        if !self.context.borrow().is_none() {
+            return Err(());
```

Comment:
I think this will throw a "DataCloneError" DOMException, but I think this has to be handled at: https://github.com/servo/servo/blob/fcb2a4cd95a19c8ef43f73c1c116117785a61cc4/components/script/dom/bindings/structuredclone.rs#L602. Would be good for a follow-up by adding a TODO, and filing an issue(also please contradict if I'm wrong and you have a test passing that checks for "InvalidStateError".

Reply:
Yep, I filled new task to fix it